app/src/main/java/com/hover/stax/presentation/simcard/SimScreen.kt

```diff
@@ -0,0 +1,333 @@
+package com.hover.stax.presentation.simcard
+
+import androidx.annotation.StringRes
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.itemsIndexed
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.shadow
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import coil.compose.AsyncImage
+import coil.request.CachePolicy
+import coil.request.ImageRequest
+import com.hover.stax.R
+import com.hover.stax.accounts.AccountsViewModel
+import com.hover.stax.bonus.BonusViewModel
+import com.hover.stax.domain.model.Account
+import com.hover.stax.presentation.home.BalanceTapListener
+import com.hover.stax.presentation.home.TopBar
+import com.hover.stax.ui.theme.*
+import com.hover.stax.utils.DateUtils
+import com.hover.stax.utils.network.NetworkMonitor
+import org.koin.androidx.compose.getViewModel
+
+data class SimScreenClickFunctions(val onClickedAddNewAccount: () -> Unit,
+                                   val onClickedSettingsIcon: () -> Unit,
+                                   val onClickedBuyAirtime: () -> Unit)
+
+@Composable
+fun SimScreen(simScreenClickFunctions: SimScreenClickFunctions,
+              balanceTapListener: BalanceTapListener) {
+	val simViewModel : SimViewModel = getViewModel()
+	val bonusViewModel: BonusViewModel = getViewModel()
+
+	val accounts = simViewModel.telecomAccounts.observeAsState(initial = null)
+	val hasNetwork by NetworkMonitor.StateLiveData.get().observeAsState(initial = false)
+	val presentSims = simViewModel.presentSims.observeAsState(initial = emptyList())
+	val bonuses = bonusViewModel.bonusList.collectAsState()
```

Comment:
There exists a usecase to fetch bonuses. Add it to `SimViewModel` and remove the `BonusViewModel`.